lib/rules/indent_style.ts

```diff
@@ -16,40 +17,39 @@ function resolve(settings: eclint.Settings) {
 	}
 }
 
-function check(context: eclint.Context, settings: eclint.Settings, line: linez.Line) {
+function check(settings: eclint.Settings, line: linez.Line) {
+	function creatErroe(message: string, columnNumber: number = 1) {
+		message = message.replace(/\b(\d+)\s.+?$/, function(str, count) {
+			if (+count > 1) {
+				str += 's';
+			}
+			return str;
+		});
+		var error = new EditorConfigError(message);
+		error.lineNumber = line.number;
+		error.columnNumber = columnNumber;
+		error.rule = 'indent_style';
+		error.source = line.text;
+		return error;
+	}
+
 	switch (resolve(settings)) {
 		case 'tab':
 			if (_.startsWith(line.text, ' ')) {
-				context.report([
-					'line ' + line.number + ':',
-					'invalid indentation: found a leading space, expected: tab'
-				].join(' '));
-				return;
+				return creatErroe('invalid indentation: found a leading space, expected: tab');
 			}
 			var softTabCount = identifyIndentation(line.text, settings).softTabCount;
 			if (softTabCount > 0) {
-				context.report([
-					'line ' + line.number + ':',
-					'invalid indentation: found ' + softTabCount + ' soft tab'
-				].join(' ') + ((softTabCount > 1) ? 's' : ''));
-				return;
+				return creatErroe('invalid indentation: found ' + softTabCount + ' soft tab');
```

Comment:
Please use a string template for this:
```ts
return createError(`invalid indentation: found ${softTabCount} soft tab(s)');
```
Notice I said "tab(s)", as the number can be singular or plural.